Assets/MRTK/StandardAssets/Shaders/MixedRealityTextMeshProSprite.shader

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// NOTE: MRTK Shaders are versioned via the MRTK.Shaders.sentinel file.
+// When making changes to any shader's source file, the value in the sentinel _must_ be incremented.
+
+// Text Mesh Pro Sprite shader with MRTK Additions
+
+// MRTK Additions
+// - Single Pass Instanced Stereo Rendering Support
+// - Support for Clipping Primitives (Plane, Sphere, Box)
+// - Added to MRTK namespace

```

Comment:
Would we be able to add these additions to https://github.com/microsoft/MixedRealityToolkit-Unity/blob/main/Assets/MRTK/StandardAssets/Shaders/Text3DShader.shader instead of adding a new shader? I see references to clipping primitives in that shader already

Reply:
The issue is that Text3DShader.shader does not support sprites though it has very similar code, and the feature request was for sprite support. I created this shader based Cameron's response here: https://github.com/microsoft/MixedRealityToolkit-Unity/issues/10045#issuecomment-877384474

Reply:
Is the diff small enough that it'd be reasonable to roll this shader into the other MRTK one? Or is it better to just leave them separate? (also a question for @Cameron-Micka)

Reply:
Great questions! The current [MRTK/TextMeshPro](https://github.com/microsoft/MixedRealityToolkit-Unity/blob/main/Assets/MRTK/StandardAssets/Shaders/MixedRealityTextMeshPro.shader) shader is based off of the [TMP/Mobile/SDF](https://github.com/UnityTechnologies/BuffaloKing/blob/main/Assets/TextMesh%20Pro/Shaders/TMP_SDF-Mobile.shader) shader. The MRTK/TextMeshPro is mostly the same as the TMP/Mobile/SDF shader sans changes needed to support single pass instanced rendering, clipping primitives, and a few other quality of life improvements. 

We did this when Text Mesh Pro was included within MRTK (before the package manager existed) so we didn't really discuss license issues with forking them. Fortunately, Unity has released all of these shaders under an MIT license [ in this repo](https://github.com/UnityTechnologies/BuffaloKing/tree/main/Assets/TextMesh%20Pro/Shaders). You can find the sprite shader these changes are based off of [here](https://github.com/UnityTechnologies/BuffaloKing/blob/main/Assets/TextMesh%20Pro/Shaders/TMP_Sprite.shader).

I think it would be a fair amount of work (but not impossible) to roll sprite support and text support into one uber MRTK/TextMeshPro shader. Text Mesh Pro takes the approach of individual shaders based on feature/platform as [you can see here](https://github.com/UnityTechnologies/BuffaloKing/tree/main/Assets/TextMesh%20Pro/Shaders). So I vote to keep the approach @elbuhofantasma has taken and have separate shaders for text and sprite support.

Reply:
I'm also in favor of keeping the two shader's separate, since the effort to merge the two seems higher than the gains we'd get from from having one shader for both sprites and text.